### PR TITLE
example simulation transportless

### PR DESCRIPTION
**Pull request by supernovahs**
Ref #4980 

I have a small question.
How should I access trace call ? 
cc @mattsse 
thanks

### CODE REVIEW

File: crates/rpc/rpc-builder/src/lib.rs [hunk 1]
```diff
@@ -717,7 +717,7 @@ pub struct RethModuleRegistry<Provider, Pool, Network, Tasks, Events> {
     /// Additional settings for handlers.
     config: RpcModuleConfig,
     /// Holds a clone of all the eth namespace handlers
-    eth: Option<EthHandlers<Provider, Pool, Network, Events>>,
+    pub eth: Option<EthHandlers<Provider, Pool, Network, Events>>,
```
Comment (mattsse) [inline crates/rpc/rpc-builder/src/lib.rs:720]: we don't want to expose internals here
```suggestion
   eth: Option<EthHandlers<Provider, Pool, Network, Events>>,
```

File: examples/trace-transaction-cli/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    /// This tells the reth CLI to install the `txpool` rpc namespace via `RethCliTxpoolExt`
+    type Node = RethCliTxpoolExt;
+}
+
+/// Our custom cli args extension that adds one flag to reth default CLI.
+#[derive(Debug, Clone, Copy, Default, clap::Args)]
+struct RethCliTxpoolExt {
+    /// CLI flag to enable the txpool extension namespace
+    #[clap(long)]
+    pub receipient: Address,
```
Comment (mattsse) [inline examples/trace-transaction-cli/src/main.rs:39]: let's do multiple
```suggestion
    #[arg(long, value_delimiter = ',')]
    pub recipients: Vec<Address>,
```

File: examples/trace-transaction-cli/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+            // Spawn an async block to listen for transactions.
+            tokio::spawn(async move {
+                // Awaiting for a new transaction and print it.
+                while let Some(tx) = tx_subscription.recv().await {
+                    println!("Transaction received: {:?}", tx);
+                    let res = provider.transaction_by_hash(tx);
+                    if let Ok(Some(tx_receipient)) = res {
+                        if tx_receipient.kind().to() == Some(receipient) {
+                            // TODO: CAll trace_call
```
Comment (mattsse) [inline examples/trace-transaction-cli/src/main.rs:70]: we need an instance of `TraceApi`:

https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc-builder/src/lib.rs#L862-L862

we can add a new function `fn trace_api` that returns this from the registry, similar to:

https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc-builder/src/lib.rs#L1078-L1081

**Review by mattsse — CHANGES_REQUESTED** [review]
this is great!

left some pointers,

we also need a way to trace a `Transaction` via:

https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc/src/trace.rs#L70-L70

we want:
a conversion function in the rpc-types-compat crate that converts a transaction into a callrequest.

File: examples/trace-transaction-cli/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        println!("RPC Server has started!");
+        let provider = rpc_components.registry.provider().clone();
+        let receipient = self.receipient;
+        if let Some(eth_handlers) = rpc_components.registry.eth.clone() {
+            let api = eth_handlers.api;
+            let tx_pool = api.pool();
+            let mut tx_subscription = tx_pool.pending_transactions_listener();
+
+            // Spawn an async block to listen for transactions.
+            tokio::spawn(async move {
```
Comment (mattsse) [inline examples/trace-transaction-cli/src/main.rs:63]: let's use the `components.tasks` to spawn the service

File: examples/trace-transaction-cli/src/main.rs [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        _components: &Reth,
+        rpc_components: RethRpcComponents<'_, Reth>,
+        _handles: RethRpcServerHandles,
+    ) -> eyre::Result<()>
+    where
+        Conf: RethRpcConfig,
+        Reth: RethNodeComponents,
+    {
+        println!("RPC Server has started!");
+        let provider = rpc_components.registry.provider().clone();
```
Comment (mattsse) [inline examples/trace-transaction-cli/src/main.rs:55]: let's use the provider from the `components`

**Comment by supernovahs** [comment]
We would have to change the. u64,u128 used in https://github.com/paradigmxyz/reth/blob/main/crates/primitives/src/transaction/mod.rs
As CallRequest uses alloy primitives .
Should we do this in a separate PR?
thx

File: examples/trace-transaction-cli/Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+name = "trace-transaction-cli"
+version = "0.0.0"
+publish = false
+edition.workspace = true
+license.workspace = true
+
+[dependencies]
+reth.workspace = true
+reth-rpc.workspace = true
+reth-primitives.workspace = true
```
Comment (mattsse) [inline examples/trace-transaction-cli/Cargo.toml:11]: all of these should be available from the reth crate as reexports

**Review by mattsse — CHANGES_REQUESTED** [review]

**Comment by mattsse** [comment]
> We would have to change the. u64,u128 used in

I don't understand, we should be able to create a CallRequest from a Transaction by simply converting the types

we need to do this in this example, because otherwise we can call the trace function?

**Comment by supernovahs** [comment]
cc @mattsse

**Review by mattsse — APPROVED** [review]
this is great, awesome work, I made some small changes:

* don't try to fetch the transaction from db (because it isn't included yet)
* use existing trait to convert a pool transaction into TransactionSignedRecovered then convert this into CallRequest
*

**Comment by supernovahs** [comment]
Too much to learn for me .thx

**Comment by mattsse** [comment]
I know :)
you'll get there, lmk if you want to continue with something bigger

btw I opened https://github.com/paradigmxyz/reth/issues/5051 which should be nice to have

**Pull request merged**